ui/test/nuxeo-layout.test.js

```diff
@@ -27,57 +35,116 @@ const base = url.substring(0, url.lastIndexOf('/'));
+      if (!isElementVisible(nuxeoError)) {
+        await waitForAttrMutation(nuxeoError, 'hidden', null);
+      }
+      expect(isElementVisible(container)).to.be.false;
+      expect(isElementVisible(nuxeoError)).to.be.true;
+    });
 
-  test('layout valid field invalid', async () => {
-    const layout = await fixture(html`
-      <nuxeo-layout href="${base}/dummy-layout.html" model='{"text": "valid"}'></nuxeo-layout>
-    `);
-    const validity = await layout.validate();
-    expect(validity).to.equal(false);
-  });
+    test('Should display error messages when layout is not found', async () => {
+      const layout = await fixture(
+        html`
+          <nuxeo-layout href="notfound.html"></nuxeo-layout>
+        `,
+      );
+      await flush();
 
-  test('layout valid field valid', async () => {
-    const layout = await fixture(html`
-      <nuxeo-layout href="${base}/dummy-layout.html" model='{"text": "valid", "required": "foo"}'></nuxeo-layout>
-    `);
-    const validity = await layout.validate();
-    expect(validity).to.equal(true);
+      const nuxeoError = layout.$$('nuxeo-error');
+      const errorCode = nuxeoError.$$('div.code');
+      const errorURL = nuxeoError.$$('div.url');
+      const errorMessage = nuxeoError.$$('div.message');
+
+      if (!isElementVisible(nuxeoError)) {
+        await waitForAttrMutation(nuxeoError, 'hidden', null);
+      }
+      expect(isElementVisible(errorCode)).to.be.true;
+      expect(errorCode.innerText).to.be.equal('404');
+      expect(isElementVisible(errorURL)).to.be.true;
+      expect(errorURL.innerText).to.be.equal('notfound.html');
+      expect(isElementVisible(errorMessage)).to.be.true;
+      expect(errorMessage.innerText).to.be.equal('Failed to find layout');
+    });
+
+    test('Should hide layout when href is empty', async () => {
+      const layout = await fixture(
+        html`
+          <nuxeo-layout href=""></nuxeo-layout>
+        `,
+      );
+      expect(isElementVisible(layout)).to.be.false;
+    });
   });
 
-  test('layout invalid field valid', async () => {
-    const layout = await fixture(html`
-      <nuxeo-layout href="${base}/dummy-layout.html" model='{"text": "invalid", "required": "foo"}'></nuxeo-layout>
-    `);
-    const validity = await layout.validate();
-    expect(validity).to.equal(false);
+  suite('Stamping', () => {
+    test('Should stamp the layout when model is defined', async () => {
+      const layout = await fixture(html`
+        <nuxeo-layout href="${base}/dummy-layout.html" model='{"text": "dummy"}'></nuxeo-layout>
+      `);
+      await flush();
+
+      if (!layout.element) {
+        await waitChanged(layout, 'element');
+      }
+
+      const span = layout.element.$$('span');
+      expect(isElementVisible(layout.$$('nuxeo-error'))).to.be.false;
+      expect(layout.element.localName).to.be.equal('dummy-layout');
+      expect(span.innerText).to.exist.and.to.be.equal('dummy');
+    });
```

Comment:
same proposition as above, about declaring these constants closer to their usage